app/services/message_templates/hook_execution_service.rb

```diff
@@ -5,18 +5,22 @@ def perform
     return if inbox.agent_bot_inbox&.active?
     return if conversation.campaign.present?
 
-    # TODO: let's see whether this is needed and remove this and related logic if not
-    # ::MessageTemplates::Template::OutOfOffice.new(conversation: conversation).perform if should_send_out_of_office_message?
-
-    ::MessageTemplates::Template::Greeting.new(conversation: conversation).perform if should_send_greeting?
-    ::MessageTemplates::Template::EmailCollect.new(conversation: conversation).perform if inbox.enable_email_collect && should_send_email_collect?
+    trigger_templates
   end
 
   private
 
   delegate :inbox, :conversation, to: :message
   delegate :contact, to: :conversation
 
+  def trigger_templates
+    # TODO: let's see whether this is needed and remove this and related logic if not
+    # ::MessageTemplates::Template::OutOfOffice.new(conversation: conversation).perform if should_send_out_of_office_message?
+    ::MessageTemplates::Template::Greeting.new(conversation: conversation).perform if should_send_greeting?
+    ::MessageTemplates::Template::EmailCollect.new(conversation: conversation).perform if inbox.enable_email_collect && should_send_email_collect?
+    ::MessageTemplates::Template::CsatSurvey.new(conversation: conversation).perform if should_send_csat_survey?
```

Comment:
In my opinion, this shouldn't run on every message. It should be an event on the conversation after_commit hook.

Reply:
Our conversation model is already pretty bloated. Hence the thought process to keep all the message templates in a single place for maintainability. Performance impact should be minimal since we already have the conversation object in memory. 

Let's refactor the execution service in a future iteration to run on conversation commit hooks as well.